File: Python/Problem-13.py
# Write A Program To Find The Factorial Of Number

# number = int(input("Enter Any Number: "))
# fact = 1
# if number < 0:
#   print("Factorial Is Does not Exist")
# elif number == 0:
#   print(f"{number} Factorial Is 1")
# else:
#   for i in range(1 , number+1):
#     fact = fact * i
#   print(fact)

# Using Recoursion
def factorial(n):
  if n < 0:
    return 0
  elif n == 0:
    return 1
  else:
    return n * factorial(n -1)
number = int(input("Enter Any number: "))
print(factorial(number))

File: JavaScript/Problem-13.js
// Write a Program To Find the Factorial Of a Number
const prompt = require("prompt-sync")();

// const number = parseInt(prompt("Enter Any Number: "));
// let fact = 1;
// if (number < 0) {
//   console.log("Factorial is not Exist");
// } else if (number === 0) {
//   console.log(`${0} factorial is 1`);
// } else {
//   for (let i = 1; i <= number; i++) {
//     fact = fact * i;
//   }
//   console.log(fact);
// }

// Using Recousion
const factorial = (num) => {
  if (num == 0) {
    return 1;
  } else if (number < 0) {
    return 0;
  } else {
    return num * factorial(num - 1);
  }
};
const number = parseInt(prompt("Enter Any number: "));
console.log(factorial(number));
